tx/textual/valuerenderer/coin_test.go

```diff
@@ -77,10 +79,27 @@ func TestCoinJsonTestcases(t *testing.T) {
 				require.NoError(t, err)
 				require.Equal(t, 1, len(screens))
 				require.Equal(t, tc.Text, screens[0].Text)
-			}
 
-			// TODO Add parsing tests
-			// https://github.com/cosmos/cosmos-sdk/issues/13153
+				var baseDenom string
+				if tc.Metadata != nil {
+					baseDenom = tc.Metadata.Display
+				} else {
+					baseDenom = strings.Split(screens[0].Text, " ")[1]
+				}
+
+				ctx = context.WithValue(context.Background(), mockCoinMetadataKey(baseDenom), tc.Metadata)
```

Comment:
Let's remove this in favor of adding this on L71:
```go
if tc.Metadata != nil {
	ctx = context.WithValue(ctx, mockCoinMetadataKey(tc.Metadata.Display), tc.Metadata)
}
```

Reply:
this case is added to handle few cases which are having metadata.display as empty values.

Reply:
Yes, I understand, I'm proposing to regroup all `ctx` things into L71:
```diff
// L71
ctx := context.WithValue(context.Background(), mockCoinMetadataKey(tc.Proto.Denom), tc.Metadata)
+ if tc.Metadata != nil {
+	ctx = context.WithValue(ctx, mockCoinMetadataKey(tc.Metadata.Display), tc.Metadata)
+ }
```

Reply:
moved up